modin/engines/dask/pandas_on_dask/frame/partition_manager.py

```diff
@@ -151,30 +104,20 @@ def broadcast_apply(cls, axis, apply_func, left, right, other_name="r"):
             NumPy array of result partition objects.
         """
 
-        def mapper(df, others):
+        def map_func(df, *others):
             other = pandas.concat(others, axis=axis ^ 1)
             return apply_func(df, **{other_name: other})
 
-        client = default_client()
+        map_func = cls.preprocess_func(map_func)
+        rt_axis_parts = cls.axis_partition(right, axis ^ 1)
         return np.array(
             [
                 [
-                    PandasOnDaskFramePartition(
-                        client.submit(
-                            deploy_func,
-                            part.future,
-                            mapper,
-                            part.call_queue,
-                            [obj[col_idx].call_queue for obj in right]
-                            if axis
-                            else [obj.call_queue for obj in right[row_idx]],
-                            *(
-                                [obj[col_idx].future for obj in right]
-                                if axis
-                                else [obj.future for obj in right[row_idx]]
-                            ),
-                            pure=False,
-                        )
+                    part.apply(
+                        map_func,
+                        *rt_axis_parts[col_idx].list_of_blocks
+                        if axis
+                        else rt_axis_parts[row_idx].list_of_blocks,
```

Comment:
if the star ("unpacking a tuple") is applied to the whole `a if foo else b` statement I suggest wrapping it in braces for readability... if it's applied to only "true" branch then why such asymmetry?

Reply:
Yes, the star should be applied to the whole statement. Will be fixed.

Reply:
Fixed